cairo/src/utils/hashdict.cairo

```diff
@@ -0,0 +1,28 @@
+from starkware.cairo.common.cairo_builtins import PoseidonBuiltin
+from starkware.cairo.common.builtin_poseidon.poseidon import poseidon_hash
+from starkware.cairo.common.dict_access import DictAccess
+from starkware.cairo.common.squash_dict import squash_dict
+
+from ethereum_types.numeric import U256
+from ethereum_types.bytes import Bytes32
+
+// A Wrapper around dict functions that hashes the key before accessing the dictionary.
+
+func hashdict_bytes32_read{poseidon_ptr: PoseidonBuiltin*, dict_ptr: DictAccess*}(key: Bytes32) -> (
+    value: felt
+) {
+    alloc_locals;
+    local value;
+    let (hashed_key) = poseidon_hash(key.value.low, key.value.high);
```

Comment:
use `poseidon_hash_many` and enable for any length bytes?

Reply:
no, we only have bytes32 and address keys. that's overkill